File: @LV1/1.5/ptice/ptice.go
package main

import "fmt"

var a, b, g []rune

func max(score []int) int {
	m := -1
	for i := 0; i < len(score); i++ {
		if score[i] > m {
			m = score[i]
		}
	}
	return m
}

func genAns() {
	for i := 0; i < 40; i++ {
		a = append(a, []rune("ABC")...)
		b = append(b, []rune("BABC")...)
		g = append(g, []rune("CCAABB")...)
	}
}

func main() {
	genAns()
	var n int
	var exam string
	score := []int{0, 0, 0}
	name := []string{"Adrian", "Bruno", "Goran"}

	fmt.Scanf("%d", &n)
	fmt.Scanln(&exam)

	for i, ch := range exam {
		if a[i] == ch {
			score[0]++
		}
		if b[i] == ch {
			score[1]++
		}
		if g[i] == ch {
			score[2]++
		}
	}
	m := max(score)

	fmt.Println(m)
	for i := 0; i < 3; i++ {
		if score[i] == m {
			fmt.Println(name[i])
		}
	}
}

File: @LV1/1.5/ptice/ptice.rs
use std::io;

fn repeat_string(s: &str, n: usize) -> Vec<char> {
    let tmp = s.repeat(n);
    return tmp.chars().collect::<Vec<char>>();
}

fn main() {
    let a = repeat_string("ABC", 40);
    let b = repeat_string("BABC", 25);
    let g = repeat_string("CCAABB", 20);

    let mut score = vec![0, 0, 0];
    let name = vec!["Adrian", "Bruno", "Goran"];

    let stdin = io::stdin();
    let mut buff = String::new();
    stdin.read_line(&mut buff).unwrap();

    let mut buff = String::new();
    stdin.read_line(&mut buff).unwrap();

    for (i, ch) in buff.trim().chars().enumerate() {
        if a[i] == ch {
            score[0] += 1;
        }
        if b[i] == ch {
            score[1] += 1;
        }
        if g[i] == ch {
            score[2] += 1;
        }
    }
    let m = *score.iter().max().unwrap();

    println!("{}", m);

    for i in 0..3 {
        if score[i] == m {
            println!("{}", name[i]);
        }
    }
}
